server.js

```diff
@@ -14,10 +15,60 @@ const welcomeMessage = {
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+let messages = require("./messages.json");
 
-app.get("/", function (request, response) {
-  response.sendFile(__dirname + "/index.html");
+app.get("/", (req, res) => {
+  res.sendFile(__dirname + "/index.html");
+});
+
+app.get("/messages", (req, res) => {
+  res.send(messages);
+});
+
+app.get("/messages/search", (req, res) => {
+  const searchText = req.query.text ? req.query.text.toLowerCase() : null;
+  const searchFrom = req.query.from.toLowerCase();
+  const searchMsgs = messages.filter(
+    (message) =>
+      message.text.toLowerCase().includes(searchText) ||
+      message.from.toLowerCase().includes(searchFrom)
+  );
```

Comment:
This will only work if both text and from are used as query parameters - if you only provide one, then an error is thrown, which I don't think this is correct behaviour.